lib/core/engine.js

```diff
@@ -210,11 +210,17 @@ class Engine {
         if (err) {
           return cb({name: "No Blockchain node found", status: 'off'});
         }
-        return cb({
-          //name: (self.web3.version.node.split("/")[0] + " " + self.web3.version.node.split("/")[1].split("-")[0] + " (Ethereum)"),
-          // TODO: get nodename
-          name: "Ethereum node",
-          status: 'on'
+
+        // TODO: web3_clientVersion method is currently not implemented in web3.js 1.0
+        self.web3._requestManager.send({method: 'web3_clientVersion', params: []}, (err, version) => {
+          if (err) {
+            return cb({name: "Ethereum node (version unknown)", status: 'on'});
+          }
+          let nodeName = version.split("/")[0];
+          let versionNumber = version.split("/")[1].split("-")[0];
```

Comment:
You could put `version.split("/")` in a const above just so you don't have to split twice, but it's pretty insignificant so no pressure to do it.

Reply:
Although I agree the extra var would avoid the extra split, I feel the extra declaration is not worth it, even clarity wise, in this case at least